app/models/extraction_definition.rb

```diff
@@ -80,4 +83,17 @@ def clone(pipeline, name)
 
     cloned_extraction_definition
   end
+
+  def destroy_associated_definitions
+    # Remove all associated harvest reports
+    extraction_jobs.each do |extraction_job|
+      next if extraction_job.harvest_job.blank?
+
+      extraction_job.harvest_job.harvest_report.destroy!
```

Comment:
question: would it be possible to have delete cascade of associated harvest report of a harvest job? Perhaps we don't want this behaviour for some reason